primefaces/src/main/java/org/primefaces/component/autocomplete/AutoComplete.java

```diff
@@ -169,7 +169,13 @@ public void broadcast(javax.faces.event.FacesEvent event) throws javax.faces.eve
                             .order(SortOrder.ASCENDING)
                             .build());
             suggestions = lazyModel.load(0, getMaxResults(), sortBy, searchFilter);
-            suggestionsCount = lazyModel.count(searchFilter);
+            List suggestionsList = (List) suggestions;
+            if (suggestionsList.size() < getMaxResults()) {
```

Comment:
We should put getMaxResults into a local variable so the getter is not called multiple times.